### PR TITLE
Feat: Add stardust basic output details and update to latest iota.js version

### PR DESCRIPTION
**Pull request by abbasabid**
# Description of change

- Output details implementation on message page for stardust network (with BasicOutput)
- Update MilestonePayload to latest iota.js interface
- Refactored transaction helper

Aims to complete task https://github.com/iotaledger/explorer/issues/262

Aims to complete https://github.com/iotaledger/explorer/issues/258

## Type of change

- Stardust upgrade (breaking-change)

## How the change has been tested

- Visually

## Change checklist

- [x] My code follows the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: client/src/app/components/stardust/FeatureBlock.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+/**
+ * Component which will display an Feature Block.
+ */
+class FeatureBlock extends AsyncComponent<FeatureBlockProps, FeatureBlockState> {
+    constructor(props: FeatureBlockProps) {
+        super(props);
+
+        this.state = {
+            showOutputDetails: -1
```
Comment (msarcev) [inline client/src/app/components/stardust/FeatureBlock.tsx:19]: Can't we just use a boolean ?

Comment (msarcev) [reply client/src/app/components/stardust/FeatureBlock.tsx:19]: Nevermind, i see.

File: client/src/app/components/stardust/MilestonePayload.tsx [hunk 2]
```diff
@@ -144,7 +144,7 @@ class MilestonePayload extends AsyncComponent<MilestonePayloadProps, MilestonePa
                                 </div>
                             </React.Fragment>
                         )}
-                        {this.props.payload.signatures && (
+                        {/* {this.props.payload.signatures && (
```
Comment (msarcev) [inline client/src/app/components/stardust/MilestonePayload.tsx:147]: Remove commented code.

File: client/src/app/routes/stardust/Addr.tsx [hunk 3]
```diff
@@ -738,7 +738,7 @@ class Addr extends AsyncComponent<RouteComponentProps<AddrRouteProps>, AddrState
                 this._bechHrp,
                 this._tangleCacheService);
         const inputsRelated = inputs.filter(input => input.transactionAddress.hex === this.state.address);
-        const outputsRelated = outputs.filter(output => output.address.hex === this.state.address);
+        const outputsRelated = outputs.filter(output => output);
```
Comment (msarcev) [inline client/src/app/routes/stardust/Addr.tsx:741]: This looks weird now. If this is filter predicate, then we don't need to filter i guess.

File: client/src/helpers/stardust/transactionsHelper.ts [hunk 4]
```diff
@@ -20,7 +20,11 @@ export interface Input {
 }
 
 export interface Output {
-    index: number; type: 2 | 3 | 4 | 5 | 6; address: IBech32AddressDetails; amount: number; isRemainder: boolean;
+    index: number;
+    type: number;
```
Comment (msarcev) [inline client/src/helpers/stardust/transactionsHelper.ts:24]: Why number? I think is better if we constrict it to possible types.

**Review by msarcev — CHANGES_REQUESTED** [review]

File: client/src/app/components/stardust/FeatureBlock.tsx (hunk 1 above)
Comment (abbasabid) [reply client/src/app/components/stardust/FeatureBlock.tsx:19]: ignored

File: client/src/app/components/stardust/MilestonePayload.tsx (hunk 2 above)
Comment (abbasabid) [reply client/src/app/components/stardust/MilestonePayload.tsx]: Done.

File: client/src/app/routes/stardust/Addr.tsx (hunk 3 above)
Comment (abbasabid) [reply client/src/app/routes/stardust/Addr.tsx]: Changed filter to map js function

File: client/src/helpers/stardust/transactionsHelper.ts (hunk 4 above)
Comment (abbasabid) [reply client/src/helpers/stardust/transactionsHelper.ts]: Done.

File: client/src/app/components/stardust/NewOutputProps.tsx [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+     */
+    output: OutputTypes;
+
+    /**
+     * The amount to display.
+     */
+    amount: number;
+
+    /**
+     * Hide the label.
```
Comment (msarcev) [inline client/src/app/components/stardust/NewOutputProps.tsx:25]: Let's call it "Should hide the label"

File: client/src/app/components/stardust/NewOutputState.tsx [hunk 6]
```diff
@@ -0,0 +1,9 @@
+import { OutputTypes } from "@iota/iota.js-stardust";
+
+export interface NewOutputState {
+    output: OutputTypes;
```
Comment (msarcev) [inline client/src/app/components/stardust/NewOutputState.tsx:4]: Missing jsdoc here.

File: client/src/app/routes/stardust/Addr.tsx [hunk 7]
```diff
@@ -716,7 +716,7 @@ class Addr extends AsyncComponent<RouteComponentProps<AddrRouteProps>, AddrState
             this._tangleCacheService
         );
         const inputsRelated = inputs.filter(input => input.transactionAddress.hex === this.state.address);
-        const outputsRelated = outputs.filter(output => output.address.hex === this.state.address);
+        const outputsRelated = outputs.map(output => output);
```
Comment (msarcev) [inline client/src/app/routes/stardust/Addr.tsx:719]: What's the point of this map ? It doesn't change anything.

File: client/src/app/components/stardust/NewOutput.tsx [hunk 8]
```diff
@@ -0,0 +1,190 @@
...
+import { ReactComponent as DropdownIcon } from "./../../../assets/dropdown-arrow.svg";
+import FeatureBlock from "./FeatureBlock";
+import { NewOutputProps } from "./NewOutputProps";
+import { NewOutputState } from "./NewOutputState";
+import UnlockCondition from "./UnlockCondition";
+
+/**
+ * Component which will display an output.
+ */
+class NewOutput extends Component<NewOutputProps, NewOutputState> {
```
Comment (msarcev) [inline client/src/app/components/stardust/NewOutput.tsx:14]: Why is it called "New" output ? Can we just call it "Output" ?

**Review by msarcev — CHANGES_REQUESTED** [review]

File: client/src/app/components/stardust/NewOutputState.tsx (hunk 6 above)
Comment (abbasabid) [reply client/src/app/components/stardust/NewOutputState.tsx:7]: done

File: client/src/app/routes/stardust/Addr.tsx (hunk 7 above)
Comment (abbasabid) [reply client/src/app/routes/stardust/Addr.tsx]: fixed.

File: client/src/app/components/stardust/NewOutput.tsx (hunk 8 above)
Comment (abbasabid) [reply client/src/app/components/stardust/NewOutput.tsx:14]: There is already a component in the folder named Output that's why i named it NewOutput. You can suggest a better name and we can call it that.

**Review by msarcev — APPROVED** [review]

**Pull request merged**